### PR TITLE
WIP: EPMRPP-35548 Attachment modal

### PR DESCRIPTION
**Pull request by ypetya**
- code view modal
- har file view modal

### CODE REVIEW

File: app/src/pages/inside/logPage/modals/attachmentModal/attachmentCodeModal.jsx [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+import { injectIntl, intlShape } from 'react-intl';
+import { reduxForm } from 'redux-form';
+import { ModalLayout, withModal } from 'components/main/modal';
+import Highlight from 'react-highlight';
+import { messages } from './messages';
+import './attachmentModal.scss';
+
+@withModal('attachmentCodeModal')
+@injectIntl
+@reduxForm({
```
[the remark this answers is not in the document]
Comment (ypetya) [reply app/src/pages/inside/logPage/modals/attachmentModal/attachmentCodeModal.jsx]: No use of it, removing the unneccessary forms. Thanks!

**Comment by ypetya** [comment]
Hey @DiscoElevator , can you please take another look?

File: app/src/components/main/modal/modalLayout/modalFooter/modalFooter.jsx [hunk 2]
```diff
@@ -102,7 +103,6 @@ export class ModalFooter extends Component {
               </BigButton>
             </div>
           )}
-          {customButton ? <div className={cx('button-container')}>{customButton}</div> : null}
```
Comment (DiscoElevator) [inline app/src/components/main/modal/modalLayout/modalFooter/modalFooter.jsx]: At the moment we have a case with custom button on the right (in place of the 'ok' button). Here's my suggestion:
1. Add special prop which controls custom button placement:
Modified `customButton` prop should look like this (don't forget to rebase onto fresh `develop` to get this changes):
```
    customButton: PropTypes.oneOfType([
      PropTypes.node,
      PropTypes.shape({
        onClick: PropTypes.func,
        component: PropTypes.func,
        buttonProps: PropTypes.object,,
        left: PropTypes.bool, // <--- this is our new flag
      }),
    ]),
```
2. Add special class to handle custom button position:
```
.left {
  position: absolute;
  left: 0;
}
```
3. Turn on this class depending on this new flag:
```
className={cx('button-container', { left: customButton.left })}
```

File: app/src/pages/inside/logsPage/modals/attachmentModal/attachmentCodeModal.jsx [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+      data: { language, content },
+    } = this.props;
+    const okButton = {
+      text: intl.formatMessage(messages.close),
+      onClick: (closeModal) => closeModal(),
+    };
+
+    return (
+      <ModalLayout title={intl.formatMessage(messages.title)} okButton={okButton}>
+        <form>
```
Comment (DiscoElevator) [inline app/src/pages/inside/logsPage/modals/attachmentModal/attachmentCodeModal.jsx]: This form doesn't look necessary here

File: app/src/pages/inside/logsPage/modals/attachmentModal/attachmentImageModal.jsx [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+    const command = this.generateRotationCommand(this.state.rotationAmount);
+    const style = { transform: command };
+
+    return (
+      <ModalLayout
+        title={intl.formatMessage(messages.title)}
+        okButton={this.renderOkButton(intl, messages)}
+        customButton={this.customButton(intl, messages)}
+      >
+        <div>
```
Comment (DiscoElevator) [inline app/src/pages/inside/logsPage/modals/attachmentModal/attachmentImageModal.jsx]: Does this `div` do anything? It looks unnecessary

File: app/src/components/main/modal/modalLayout/modalFooter/modalFooter.scss [hunk 5]
```diff
@@ -40,9 +40,12 @@
   }
 }
 .buttons-block {
-  float: right;
   white-space: nowrap;
+  display: flex;
+  justify-content: space-between;
```
Comment (DiscoElevator) [inline app/src/components/main/modal/modalLayout/modalFooter/modalFooter.scss]: This code affects all modals, please revert it and refer to my commentary from above

File: app/src/pages/inside/logsPage/modals/attachmentModal/img/rotate-left-icon.svg [hunk 6]
```diff
@@ -0,0 +1,4 @@
+<svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24">
```
Comment (DiscoElevator) [inline app/src/pages/inside/logsPage/modals/attachmentModal/img/rotate-left-icon.svg]: Please move this file to `common/img`

File: app/src/pages/inside/logsPage/modals/attachmentModal/attachmentImageModal.jsx [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+    const rotationA = this.state.rotationAmount;
+    this.setState({
+      rotationAmount: rotationA - 90,
+    });
+  };
+
+  generateRotationCommand = (amount) => `rotate(${amount}deg)`;
+
+  customButton = (intl) => (
+    <input
```
Comment (DiscoElevator) [inline app/src/pages/inside/logsPage/modals/attachmentModal/attachmentImageModal.jsx]: We usually define icons in this way:
```
<div className={cx('rotate-icon'}) onClick={...}>
  {Parser(RotateImageIcon)}
</div>
```
Let's stick to the common way of doing this.

P.S. In the future, acessibility issues will be resolved by using aria-attributes, for now it's not necessary.

File: app/src/pages/inside/logsPage/modals/attachmentModal/attachmentModal.jsx [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+  render() {
+    const { intl } = this.props;
+    const okButton = {
+      text: intl.formatMessage(messages.close),
+      onClick: (closeModal) => closeModal(),
+    };
+
+    return (
+      <ModalLayout title={intl.formatMessage(messages.title)} okButton={okButton}>
+        <br />
```
Comment (DiscoElevator) [inline app/src/pages/inside/logsPage/modals/attachmentModal/attachmentModal.jsx]: What is the purpose of this empty modal? Is it a base for something in future?

File: app/src/pages/inside/logsPage/modals/attachmentModal/attachmentImageModal.jsx [hunk 9]
```diff
@@ -0,0 +1,72 @@
...
+import PropTypes from 'prop-types';
+import classNames from 'classnames/bind';
+import { injectIntl, intlShape } from 'react-intl';
+import { ModalLayout, withModal } from 'components/main/modal';
+import { messages } from './messages';
+import rotateImage from './img/rotate-left-icon.svg';
+import styles from './attachmentImageModal.scss';
+
+const cx = classNames.bind(styles);
+const classes = cx('toleft', null, {});
```
Comment (DiscoElevator) [inline app/src/pages/inside/logsPage/modals/attachmentModal/attachmentImageModal.jsx]: Not needed here. See the comment about adding custom button.

**Comment by ypetya** [comment]
We created another PR for this, fixing the branch name. #983

**Pull request closed without merging**